app/src/main/java/com/portfolio/tasky/entry/data/UserPreferencesImpl.kt

```diff
@@ -0,0 +1,30 @@
+package com.portfolio.tasky.entry.data
+
+import android.content.Context
+import com.google.gson.Gson
+import com.portfolio.tasky.R
+import com.portfolio.tasky.entry.models.AuthenticationResponse
+import javax.inject.Inject
+
+class UserPreferencesImpl @Inject constructor(val context: Context) : UserPreferences {
+
+    private val sharedPreference by lazy { context.getSharedPreferences(context.getString(R.string.app_name), Context.MODE_PRIVATE)}
+
+    override fun saveAuthenticatedUser(authenticatedUser: AuthenticationResponse?) {
+        val jsonAuthenticatedUser = Gson().toJson(authenticatedUser)
+        sharedPreference.edit().putString("authUser", jsonAuthenticatedUser).apply()
+    }
+
+    override fun getAuthenticatedUser(): AuthenticationResponse? {
+        val authenticatedUserJson = sharedPreference.getString("authUser", "")
+        return if(!authenticatedUserJson.equals("") && !authenticatedUserJson.equals("null")){
+            Gson().fromJson(authenticatedUserJson, AuthenticationResponse::class.java)
+        }else{
+            AuthenticationResponse("","","")
```

Comment:
You can return null here. Instead of an empty AuthenticationResponse

Reply:
Yes, don't be afraid to use null values. They clearly indicate that a value doesn't exist which is their job. With this value, you'd assume you actually got a response, but it's empty which will lead to bugs and unnecessary debugging